### PR TITLE
Created tests for zoom lock toggling

### PR DESCRIPTION
**Pull request by timothywang03**
Describe what your pull request does. If you can, add GIFs or images showing the before and after of your change.

### Change type

- [ ] `bugfix`
- [ ] `improvement`
- [x] `feature`
- [ ] `api`
- [ ] `other`

### Test plan

1. Define a default view frame
2. Try to change view frame scale
3. Make sure that changes can't be applied if the lock is toggled

- [x] Unit tests
- [ ] End to end tests

### Release notes

- Fixed a bug with…

### CODE REVIEW

File: packages/tldraw/src/test/ZoomTool.test.ts [hunk 1]
```diff
@@ -209,4 +209,6 @@ describe('TLSelectTool.Zooming', () => {
 		})
 		editor.expectToBeIn('zoom.idle')
 	})
+
+	it("Correctly doesn't zoom in when the zoom lock is deactivated", () => {})
```
Comment (margarethe913) [inline packages/tldraw/src/test/ZoomTool.test.ts]: Is this test meant to just be empty?

File: packages/editor/api-report.md [hunk 2]
```diff
@@ -2584,6 +2585,7 @@ export interface TLCameraMoveOptions {
 export interface TLCameraOptions {
     constraints?: TLCameraConstraints;
     isLocked: boolean;
+    isZoomLocked: boolean;
```
Comment (margarethe913) [inline packages/editor/api-report.md:2588]: Also, do we want to test that the user panning with trackpad also doesn't do anything?

**Review by margarethe913 — CHANGES_REQUESTED** [review]
LGTM pending minor comments! I like how you exhaustively tested all the different zoom functionalities, but I think we might want a test for just zooming in with trackpad/fingers on tablet

File: packages/tldraw/src/test/ZoomTool.test.ts (hunk 1 above)
Comment (jankrom) [reply packages/tldraw/src/test/ZoomTool.test.ts]: LGTM besides this comment right here. Is it meant to be empty?

File: packages/editor/api-report.md (hunk 2 above)
Comment (timothywang03) [reply packages/editor/api-report.md:2588]: i dont think thats considered zooming no?

File: packages/tldraw/src/test/ZoomTool.test.ts (hunk 1 above)
Comment (timothywang03) [reply packages/tldraw/src/test/ZoomTool.test.ts]: yeah i guess ur right

**Review by timothywang03 — COMMENTED** [review]
comments applied!

**Pull request merged**